Xamarin.Forms.Platform.iOS/Renderers/EntryRenderer.cs

```diff
@@ -315,42 +333,101 @@ void UpdateReturnType()
 
 		void UpdateCursorFromControl(NSObservedChange obj)
 		{
-			var control = Control;
-			if (_selectedTextRangeIsUpdating || control == null || Element == null)
+			if (_nativeSelectionIsUpdating || Control == null || Element == null)
 				return;
 
-			var currentSelection = control.SelectedTextRange;
-			int selectionLength = (int)control.GetOffsetFromPosition(currentSelection.Start, currentSelection.End);
-			int newCursorPosition = (int)control.GetOffsetFromPosition(control.BeginningOfDocument, currentSelection.Start);
+			var currentSelection = Control.SelectedTextRange;
+			if (currentSelection != null)
+			{
+				if (!_cursorPositionChangePending)
+				{
+					int newCursorPosition = (int)Control.GetOffsetFromPosition(Control.BeginningOfDocument, currentSelection.Start);
+					if (newCursorPosition != Element.CursorPosition)
+					{
+						_nativeSelectionIsUpdating = true;
+						ElementController?.SetValueFromRenderer(Entry.CursorPositionProperty, newCursorPosition);
+					}
+				}
+
+				if (!_selectionLengthChangePending)
+				{
+					int selectionLength = (int)Control.GetOffsetFromPosition(currentSelection.Start, currentSelection.End);
 
-			_selectedTextRangeIsUpdating = true;
-			if (newCursorPosition != Element.CursorPosition)
-				ElementController?.SetValueFromRenderer(Entry.CursorPositionProperty, newCursorPosition);
+					if (selectionLength != Element.SelectionLength)
+					{
+						_nativeSelectionIsUpdating = true;
+						ElementController?.SetValueFromRenderer(Entry.SelectionLengthProperty, selectionLength);
+					}
+				}
+			}
 
-			if (selectionLength != Element.SelectionLength)
-				ElementController?.SetValueFromRenderer(Entry.SelectionLengthProperty, selectionLength);
-			_selectedTextRangeIsUpdating = false;
+			_nativeSelectionIsUpdating = false;
```

Comment:
Can we put this concept in a try/finally  across the platforms??

```C#

try{
var currentSelection = Control.SelectedTextRange;
.....
}
finally{
_nativeSelectionIsUpdating = false;
}
```

Reply:
Is this purely defensive, or did you see an example of it failing?

Reply:
Defensive
Just my reaction to a semaphore

Reply:
Latest changes are perfect!! Moving to a separate method was a good move

I tested again on all platforms and all seems good